airflow/models/taskmixin.py

```diff
@@ -115,24 +117,39 @@ def __rlshift__(self, other: DependencyMixin | Sequence[DependencyMixin]):
         self.__rshift__(other)
         return self
 
+    @abstractmethod
+    def add_to_taskgroup(self, task_group: TaskGroup) -> None:
+        """Add the task to the given task group."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def _iter_references(obj: Any) -> Iterable[DependencyMixin]:
+        from airflow.models.baseoperator import AbstractOperator
+        from airflow.models.xcom_arg import PlainXComArg
+
+        if isinstance(obj, AbstractOperator):
+            yield obj
+        elif isinstance(obj, PlainXComArg):
+            yield obj.operator
```

Comment:
I think this should cover all XComArg instances and use `yield from obj.iter_references()` instead